### ISSUE TITLE
Feature request: Oscillator pitch offset to be +/-

### ISSUE THREAD

**Issue by riban-bw** [polluxsynth/audio-plugins#10]
Currently you can offset each oscillator by 0..72 notes (coarse) and 0..100 cents (fine). It would be advantageous to be able to detune with negative values, e.g. +/- 72 notes (coarse) and +/-100 cents (fine). The default would be 0 (centre).

**Comment by riban-bw** [comment]
> Zynthian always displays the red arc starting from the lowest value, rather than starting from zero for parameters that are bipolar, so one really has to read the actual value in the middle of the arc to aim for the center zero.

Zynthian has a min, max and mid value for each controller. It supports bipolar controllers, e.g. "Calf Monosynth Ctrls#1 Osc1 PW" which spans -1.0 to +1.0 with centre 0.0. The ttl config for this is:
```
        Port 4:
                Type:        http://lv2plug.in/ns/lv2core#ControlPort
                             http://lv2plug.in/ns/lv2core#InputPort
                Symbol:      o1_pw
                Name:        Osc1 PW
                Minimum:     -1.000000
                Maximum:     1.000000
                Default:     0.000000
                Properties:  http://lv2plug.in/ns/ext/port-props#hasStrictBounds
```

Regarding ease of implementation - that is for you to decide. I wouldn't want to get bogged down with implementation detail.

Regarding ease of use - it can feel odd to adjust a value that has a centre value but without a centre indent. This can be evident in Zynthian which uses rotary encoders with indents but sometimes scalling or range issues result in the centre spanning several indents or showing negative zero (which irks me!!!). It is common to have centre zero bipolar controls in LV2 plugins (see above example) so I don't think it is necessarily wrong but again - this is a design decision for you. I understand your reluctance but I would more easily understand and prefer a centre-zero, bipolar control than a minimum-zero control plus an offset control. The former being more intutive with the sound design option on a single note which makes it easier to try different sound designs as well as use it during performance.

**Comment by polluxsynth** [comment]
What I meant was not that it's not possible to have zero-centered LV2 controls, but that the Zynthian graphic rendering of the control value as a red arc always starts at the 7 o'clock position and increases clockwise as the parameter value is increased, regardless of where the zero value of the parameter is.

It would be nice if zero-centered values had the arc start at the 12 o'clock position, growing clockwise when the parameter is increased above zero (up to a maximum of 5 o'clock or whatever the exact maximum position is today), and growing counter-clockwise when the parameter is decreased below zero. That way, visually, when the parameter is at zero (in the middle), the red arc would be at its minimum size, which would make it easier to 'aim' for the zero position.

Perhaps there is some parameter hint that can be given that causes the graphics to behave this way, I haven't checked at all, but it at least didn't happen automagically just because a parameter has both positive and negative value extremes with zero in the middle.

**Comment by riban-bw** [comment]
Ah, yes - indeed that is a Zynthian feature request rather than an issue here. I hadn't considered it an issue because I have grown accustomed to the current behaviour. We can look at resolving the Zynthian display but I don't think that affects this mimi-d request.

**Comment by polluxsynth** [comment]
Implemented in relaase 2.0.3 .
